api/scripts/fill-skill-id-in-user-saved-tutorials.js

```diff
@@ -0,0 +1,114 @@
+const { knex } = require('../tests/test-helper');
+const UserSavedTutorial = require('../lib/domain/models/UserSavedTutorial');
+const UserSavedTutorialWithTutorial = require('../lib/domain/models/UserSavedTutorialWithTutorial');
+const knowledgeElementRepository = require('../lib/infrastructure/repositories/knowledge-element-repository');
+const tutorialDatasource = require('../lib/infrastructure/datasources/learning-content/tutorial-datasource');
+const skillDatasource = require('../lib/infrastructure/datasources/learning-content/skill-datasource');
+
+async function main() {
+  console.log('Starting filling skillId to user saved tutorials');
+
+  const userSavedTutorialsWithoutSkillId = await getAllUserSavedTutorialsWithoutSkillId();
+  const tutorials = await getAllTutorials();
+  const skills = await getAllSkills();
+
+  const tutorialsWithSkills = associateSkillsToTutorial(skills, tutorials);
+
+  for (const userSavedTutorialWithoutSkillId of userSavedTutorialsWithoutSkillId) {
+    const userSavedTutorial = associateTutorialToUserSavedTutorial(
+      userSavedTutorialWithoutSkillId,
+      tutorialsWithSkills
+    );
+    if (!userSavedTutorial.tutorial) {
+      return knex('user-saved-tutorials').delete().where({ id: userSavedTutorial.id });
+    }
+
+    const skillId = await getMostRelevantSkillId(userSavedTutorial);
+
+    if (!skillId) {
+      return;
+    }
+
+    await knex('user-saved-tutorials').update({ skillId }).where({ id: userSavedTutorial.id });
+  }
+
+  console.log('End filling skillId to user saved tutorials');
+}
+
+async function getAllUserSavedTutorialsWithoutSkillId() {
+  const userSavedTutorials = await knex('user-saved-tutorials').whereNull('skillId');
+  return userSavedTutorials.map(_toDomain);
+}
+
+function _toDomain(userSavedTutorial) {
+  return new UserSavedTutorial({
+    id: userSavedTutorial.id,
+    tutorialId: userSavedTutorial.tutorialId,
+    userId: userSavedTutorial.userId,
+    skillId: userSavedTutorial.skillId,
+  });
+}
+
+async function getAllTutorials() {
+  return tutorialDatasource.list();
+}
+
+async function getAllSkills() {
+  return skillDatasource.list();
+}
+
+function _skillHasTutorialId(skill, tutorialId) {
+  return skill.tutorialIds.includes(tutorialId);
+}
+
+function associateSkillsToTutorial(skills, tutorials) {
+  return tutorials.map((tutorial) => {
+    const skillIds = skills.filter((skill) => _skillHasTutorialId(skill, tutorial.id)).map((skill) => skill.id);
+    return {
+      ...tutorial,
+      skillIds,
```

Comment:
**question:** Est-ce qu'on ne devrait pas aussi ajouter les `learningMoreSkillIds` ?

Je pense qu'il est possible qu'un utilisateur ait sauvegardé un tuto affiché dans la section "En apprendre plus".

Reply:
Deux options IMO :
- on envoie ce premier script en étant conscient que tout ne sera pas balayé (mesurer avant/après puis faire un ou plusieurs autre(s) script(s) pour tout traiter)
- on prévoie tous les cas "à la marge" : `learningMore` mais aussi un fallback si c'est ni l'un ni l'autre ?

Par contre vu qu'on n'enregistre pas encore les `skillId` il faudra refaire tourner ce ou ces scripts jusqu'à ce qu'on les enregistre.

Reply:
Je suis pour : 
- Fixer les derniers soucis de ce script
- Supprimer le cas de la suppression
- Envoyer cette première version et ne pas le lancer
- Puis itérer dessus pour ajouter les cas "à la marge"
- Ajouter le remplissage du `skillId` via les différentes routes possible
- Le lancer uniquement après tous les cas traiter

Reply:
LGTM :)